packages/perseus/src/server-item-renderer.tsx

```diff
@@ -278,9 +283,10 @@ export class ServerItemRenderer
             questionHighlightedWidgets: withRemoved,
         });
 
-        if (this.props.apiOptions.interactionCallback) {
-            this.props.apiOptions.interactionCallback();
-        }
+        // Call the interactionCallback, if it exists, with the current user input data
+        this.props.apiOptions?.interactionCallback?.(
+            this.questionRenderer.getUserInputForWidgets(),
+        );
```

Comment:
nit: wonder if it'd be worth making a shared method, something like:

`ServerItemRenderer._handleInteraction()`